### PR TITLE
gs: support other regions than us

### PR DESCRIPTION
**Pull request by Sohalt**
What does this PR change? What problem does it solve?
-----------------------------------------------------

Add an option to set the region for a bucket in google cloud storage.

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- [ ] I have added documentation for relevant changes (in the manual).
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [ ] I'm done! This pull request is ready for review.

### CODE REVIEW

File: internal/backend/gs/config.go [hunk 1]
```diff
@@ -16,13 +16,15 @@ type Config struct {
 	Bucket    string
 	Prefix    string
 
-	Connections uint `option:"connections" help:"set a limit for the number of concurrent connections (default: 5)"`
+	Connections uint   `option:"connections" help:"set a limit for the number of concurrent connections (default: 5)"`
+	Region      string `option:"region" help:"region for the bucket (default: us)"`
```
Comment (MichaelEischer) [inline internal/backend/gs/config.go:20]: From what I understand, the region is only used when creating a bucket. That should be reflected in the help text.

**Review by MichaelEischer — CHANGES_REQUESTED** [review]
The code is fine, except for the one remark below. Please also add a short changelog entry and a short remark in https://github.com/restic/restic/blob/master/doc/030_preparing_a_new_repo.rst#google-cloud-storage that it's now possible to configure the region in which a new bucket is created.

**Comment by MichaelEischer** [comment]
@Sohalt  are you still interested in finishing this PR?

**Comment by Sohalt** [comment]
Sorry for taking so long. I hope it's fine like this.

**Review by MichaelEischer — APPROVED** [review]
LGTM. Thanks!

**Comment by MichaelEischer** [comment]
I've rebased the PR and fixed the test failures.

**Pull request merged**